src/mongocxx/options/client_session.cpp

```diff
@@ -26,7 +26,20 @@ client_session& client_session::causal_consistency(bool causal_consistency) noex
 }
 
 bool client_session::causal_consistency() const noexcept {
-    return _causal_consistency;
+    // Unless causal consistency has been explicitly disabled (i.e. it has a value, and
+    // that value is false), we always return true. If snapshot reads are enabled,
+    // the invalid setting combination will later be rejected:
+    return _causal_consistency.value_or(true);
```

Comment:
It seems ambiguous to me (and frankly I have-- as is surely apparent-- struggled to understand the wording of this particular spec-- it does not say that the alternative is immediately false. In the current implementation, as the comment notes, we punt (especially in the case of snapshot being enabled). The user can set it explicitly if they need one behavior or the other; so, I think it will be correctly handled by the layer below us. On the other hand, I can see it interpreted as the case you're describing, so if seems /clear/ to you I'm entirely ok changing it.

Reply:
It is not completely clear to me either. Resolving.